### PR TITLE
Fix hess_spectrum_analysis.ipynb once and for all

### PR DESCRIPTION
**Pull request by joleroi**
This PR

* add a test for ``gammapy-extra/notebooks/hess_spectrum_analysis``
* fixes several bugs in ``SpectrumObservation.stack``

### CODE REVIEW

**Comment by joleroi** [comment]
@cdeil If you want to give feedback, please execute https://github.com/gammapy/gammapy-extra/blob/master/notebooks/hess_spectrum_analysis.ipynb and tell me about the experience as a user

It's a lot of info, but right now this is the place where stuff is explained/showcased

File: gammapy/spectrum/observation.py [hunk 1]
```diff
@@ -388,28 +386,34 @@ def stack(cls, obs_list, group_id=None):
...
-        stacked_backscal_on = np.nan_to_num(backscal_on / stacked_off_counts)
-        stacked_backscal_off = np.nan_to_num(backscal_off / stacked_off_counts)
+
+        stacked_backscal_on = backscal_on / stacked_off_counts
+        stacked_backscal_off = backscal_off / stacked_off_counts
+
+        # there should be no nan values in backscal_on or backscal_off
+        idx = np.where(stacked_off_counts == 0)[0]
+        stacked_backscal_on[idx] = -1
+        stacked_backscal_off[idx] = -1
```
Comment (cdeil) [inline gammapy/spectrum/observation.py]: Maybe make this magic fill value a class-level private constant?

These things have a tendency of getting different values in different places and causing bugs, and introducing constants with a one-line comment sometimes helps a bit maintain the code and debug issues.

**Comment by cdeil** [comment]
I don't have time for a detailed review.
We have so much crappy docs / notebooks, let's merge this without much review and move on.

You start with "Load Data" and end with "Stack observations".
I would prefer if every notebook starts with "Introduction" and ends with sometimes "Summary" and "What next?".
The intro should give an overview of what will be done, with a link to the gammapy.spectrum docs.
The what next could be pointers to the API docs for the main classes used, or suggestions what other notebooks / docs / things people might try.

Each can just be a short paragraph with two or three sentences or bullet points, but I think it's good to have this intro and outro.

**Comment by joleroi** [comment]
I agree, will change :+1:

**Comment by cdeil** [comment]
For the SED plot at the end: I always prefer `E^2 * F` and the most common is to use `erg / cm^2 / s`.
The plot could be made a bit better: have a small margin between main plot and residuals, add energy-bin-width indicators to residual panel also.

**Comment by cdeil** [comment]
Wait ... there's no Sherpa on Python 3 yet (see #686).

We should add separate Python 2 and 3 builds that run the notebooks.

And some mechanism so that it's possible to (for now) run each one only in 2 or 3.
Can this be done with the Makefile?
Or do we need to introduce an extra bash or Python script?

**Comment by joleroi** [comment]
Let's see if that works ..

**Comment by cdeil** [comment]
@joleroi - Please rebase.

This PR seems to include several changes to spectrum code and travis-ci setup.

My suggestion would be that we merge this ASAP and then I can try and fix the travis-ci issue (if any?) in a separate PR.

It's OK if we break master for a few hours, although maybe you could just uncomment the travis-ci build that breaks for now?

**Comment by joleroi** [comment]
> This PR seems to include several changes to spectrum code and travis-ci setup.

Yes, it's main purpose is to add a notebook test on travis-ci. We can however merge it (to make the spectrum notebook work) and try to fix the test issue later.

**Pull request merged**